rcl/include/rcl/guard_condition.h

```diff
@@ -0,0 +1,161 @@
+// Copyright 2015 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCL__GUARD_CONDITION_H_
+#define RCL__GUARD_CONDITION_H_
+
+#if __cplusplus
+extern "C"
+{
+#endif
+
+#include "rcl/node.h"
+#include "rcl/visibility_control.h"
+
+/// Internal rcl guard condition implementation struct.
+struct rcl_guard_condition_impl_t;
+
+/// Handle for a rcl guard condition.
+typedef struct rcl_guard_condition_t
+{
+  struct rcl_guard_condition_impl_t * impl;
+} rcl_guard_condition_t;
+
+/// Options available for a rcl guard condition.
+typedef struct rcl_guard_condition_options_t
+{
+  /// Custom allocator for the guard condition, used for internal allocations.
+  rcl_allocator_t allocator;
+} rcl_guard_condition_options_t;
+
+/// Return a rcl_guard_condition_t struct with members set to NULL.
+RCL_PUBLIC
+rcl_guard_condition_t
+rcl_get_zero_initialized_guard_condition();
```

Comment:
I see, it currently doesn't perform any complex initialization. Do you expect it to do that in the future? If not I would rather not add a function like this to our core API - simply because the caller code can do exactly the same using a few characters (`= {0}`).